### ISSUE TITLE
DatetimeRange does not support date strings with pydantic <= 2.6.0

### ISSUE THREAD

**Issue by Waffleboy** [qdrant/qdrant-client#563]
The docs [here](https://qdrant.tech/documentation/concepts/filtering/#datetime-range) states that str type datetime in RFC3339 is accepted.

<img width="623" alt="image" src="https://github.com/qdrant/qdrant-client/assets/12254813/33bfba46-e7ba-4dc1-85a0-a7b41d89471d">

However,  the following fails with the error  `pydantic.error_wrappers.ValidationError: 2 validation errors for DatetimeRange` as the DatetimeRange pydantic object expects a date object instead of a str.

```python
*** pydantic.error_wrappers.ValidationError: 2 validation errors for DatetimeRange
gte
  invalid datetime format (type=value_error.datetime)
lte
  invalid datetime format (type=value_error.datetime)
```

## Current Behavior

This is because the pydantic object defined only allows datetime instead of str as the documentation states.
```python

class DatetimeRange(BaseModel, extra="forbid"):
    """
    Range filter request
    """

    lt: Optional[datetime] = Field(default=None, description="point.key &lt; range.lt")
    gt: Optional[datetime] = Field(default=None, description="point.key &gt; range.gt")
    gte: Optional[datetime] = Field(default=None, description="point.key &gt;= range.gte")
    lte: Optional[datetime] = Field(default=None, description="point.key &lt;= range.lte")
```
## Steps to Reproduce
```python
from qdrant_client.http import models
start_date = '2024-01-01'
end_date = '2024-02-01'
 date_condition = models.FieldCondition(
            key="date_col",
            range= models.DatetimeRange(gte=start_date, lte=end_date)
        )
``` 

## Expected Behavior
Should not fail with type error

## Possible Solution
Change pydantic

## Context (Environment)
Unable to use

**Comment by joein** [comment]
Hey @Waffleboy 

Thanks for pointing it out

We'll look into it

**Comment by joein** [comment]
@Waffleboy 

it works for pydantic>=2.6.0, however fails for the other versions
The reason is that if a field has `datetime` parameter type, `date` objects can't be parsed

However, examples from the documentation works for all the versions of pydantic, since there we use datetime strings, not just date. E.g. `"2024-01-31 10:14:31Z"`

**Comment by joein** [comment]
#572

**Comment by joein** [comment]
fixed in #575

**Comment by joein** [comment]
Available as of `qdrant-client==1.9.0`
